gbpservice/nfp/lifecycle_manager/drivers/lifecycle_driver_base.py

```diff
@@ -0,0 +1,440 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from oslo_log import log as logging
+
+from gbpservice.nfp._i18n import _
+from gbpservice.nfp.lifecycle_manager.openstack import (
+    openstack_driver
+)
+
+LOG = logging.getLogger(__name__)
+
+
+class LifeCycleDriverBase(object):
```

Comment:
Should this derive from an abstract class which doesn't have any impl. This will make it easier for new drivers (such as haproxy) to see what can be overridden.